test/groupBy.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('groupBy', function() {
+
+  eq(typeof S.groupBy, 'function');
+  eq(S.groupBy.length, 2);
+  eq(S.groupBy.toString(), 'groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)');
+
+  var productsOf3 = S.curry2(function(a, b) {
+    return a * b % 3 === 0;
+  });
+
+  eq(S.groupBy(productsOf3, []), []);
+  eq(S.groupBy(productsOf3, [1, 2, 3, 4, 5, 6, 7, 8, 9]), [[1], [2, 3], [4], [5, 6], [7], [8, 9]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 3, 3, 4]), [[1, 1], [2], [3, 3], [4]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 1, 1]), [[1, 1], [2], [1, 1]]);
+
+});
```

Comment:
Ok I'll drop the number! And yeah it does the job, here's a section of the `console.log(xs, f(1)(2))`:

```
 ] false
[ 18, 15 ] false
[ 18, 12, 27, 39 ] false
[ 23, 23, 26, 32, 0 ] true
[] true
[ 45, 34 ] false
[ 33, 7, 11 ] false
[ 25, 3, 16, 42, 30, 0 ] false
[] true
[ 25, 12 ] true
[ 14, 17, 7 ] true
[ 25, 12, 5, 15, 19 ] false
[ 4, 1 ] true
[ 40, 22 ] false
[ 4 ] true
[ 1 ] false
[ 38, 27 ] false
[] true
[] true
[ 4, 5, 1 ] true
[ 10, 11, 16, 14 ] true
[] true
[ 18, 13, 20 ] true
[ 0 ] false
[ 1, 0, 10 ] true
[ 8, 12, 5, 4 ] true
[ 3, 8, 12, 33, 33 ] true
[ 32, 35, 13, 7, 48, 27 ] false
[ 2, 3, 7 ] true
[] false
[ 22, 8, 27, 14 ] false
[ 14, 7, 1, 2 ] false
[] true
[ 25, 17, 8 ] true
[ 1, 10 ] true
```